gmso/core/topology.py

```diff
@@ -630,17 +581,16 @@ def add_subtopology(self, subtop, update=True):
 
     def is_typed(self, updated=False):
         """Verify if the topology is parametrized."""
-        if not updated:
-            self.update_connection_types()
-            self.update_atom_types()
-
-        if len(self.atom_types) > 0 or len(self.connection_types) > 0:
-            self._typed = True
-        else:
-            self._typed = False
-        return self._typed
-
-    def is_fully_typed(self, updated=False, group="topology"):
+        try:
+            next(self.atom_types)
+        except StopIteration:
+            try:
+                next(self.connection_types)
+            except StopIteration:
+                return False
+        return True
+
+    def is_fully_typed(self, group="topology"):
         """Check if the topology or a specifc group of objects that make up the topology are fully typed
```

Comment:
"""Check if the topology or a specific group of objects that make up the topology are fully typed